docs/operator-manual/upgrading/2.4-2.5.md

```diff
@@ -42,6 +42,21 @@ If you rely on out of bounds symlinks, this check can be disabled one of three w
 
 It is **strongly recommended** to leave this check enabled. Disabling the check will not allow _all_ out-of-bounds symlinks. Those will still be blocked for things like values files in Helm charts, but symlinks which are not explicitly blocked by other checks will be allowed.
 
+## Deprecated client-side manifest diffs
+
+When using `argocd app diff --local`, code from the repo server is run on the user's machine in order to locally generate manifests for comparing against the live manifests of an app. However, this requires that the necessary tools (Helm, Kustomize, etc) are installed with the correct versions. Even worse, it does not support Config Management Plugins (CMPs) whatsoever.
+
+In order to support CMPs and reduce local requirements, we have implemented *server-side generation* of local manifests via the `--server-side-generate` argument. For example, `argocd app diff --local repoDir --server-side-generate` will upload the contents of `repoDir` to the repo server and run your manifest generation pipeline against it, the same as it would for a Git repo.
+
+In v2.6, the `--server-side-generate` argument will become the default and client-side generation will be removed.
+
+!!! warning
+    The semantics of *where* Argo will start generating manifests within a repo has changed between client-side and server-side generation. With client-side generation, the application's path was ignored and the value of `--local-repo-root` was effectively used (by default `/` relative to `--local`).
+    
+    For example, given an application that has an application path of `/manifests`, you would need to run `argocd app diff --local yourRepo/manifests`. This behavior did not match the repo server's process of downloading the full repo/chart and then beginning generation in the path specified in the application manifest. This would be especially problematic with repos that have symlinks pointing outside of the app path.
+
+    When using server-side generation, `--local` should be the root of your repo rather than the expected appPath. This is especially important to keep in mind when `--server-side-generate` becomes the default in v2.6.
```

Comment:
Really great writeup. Want to add a sentence to mention that the app path is inferred from the app config, i.e. `spec.source.path`?

Reply:
thank you! 😄 
done!